### ISSUE TITLE
DOMTokenList.add bug that allows adding tokens that contains HTML space characters

### ISSUE THREAD

**Issue by ahmadigbaryia** [capricorn86/happy-dom#29]
I have a function that uses the classList to add/remove class names.
I mistakenly added a two classes as one string with a space to separate them, the test PASSED but in production it failed:
"Uncaught DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided ('a b') contains HTML space characters, which are not valid in tokens"
```js
//func.js
export function applyClassName({
	oldValue,
	newValue,
	element,
	defaultValue = ""
}) {
	if (newValue) {
		if (oldValue) {
			element.classList.remove(oldValue);
		}
		element.classList.add(newValue);
	} else if (oldValue) {
		element.classList.remove(oldValue);
		element.classList.add(defaultValue);
	}
}
```
```js
//func.test.js
import { Window } from "happy-dom";
import { applyClassName } from "./func";

const window = new Window();
const document = window.document;

test.only("Test space within class name", () => {
	const myDiv = document.createElement("div");
	applyClassName({newValue: "a b", element: myDiv});
	expect(myDiv.className).toBe("a b");
});
```
I expect the test to fail because it will throw an exception, but using happy-dom it dosen't, it passes.

**Comment by capricorn86** [comment]
Hi @ahmadigbaryia!

Thank you for filing this bug. I will fix it as soon as I can.

**Comment by capricorn86** [comment]
@ahmadigbaryia, I have now made a check for spaces in the class string. It will throw an error if it contains spaces.

The fix is available in v0.11.1 of happy-dom and v. 0.0.40 of jest-environment-happy-dom.
